packages/teleport/src/Nodes/AddNode/Iam/Iam.tsx

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import { DialogContent, DialogFooter } from 'design/Dialog';
+import { Text, Box, ButtonPrimary, Link, Alert, ButtonSecondary } from 'design';
+import TextSelectCopy from 'teleport/components/TextSelectCopy';
+import FieldInput from 'shared/components/FieldInput';
+import Validation, { Validator } from 'shared/components/Validation';
+import { Attempt } from 'shared/hooks/useAttemptNext';
+import { Rule } from 'teleport/services/joinToken';
+import { createBashCommand } from '../useAddNode';
+
+export default function Iam({ token, attempt, onGenerate, onClose }: Props) {
+  const [rule, setRule] = React.useState<Rule>({
+    awsAccountId: '',
+    awsArn: '',
+  });
+
+  function handleGenerate(validator: Validator) {
+    // validate() will run the rule functions of the form inputs
+    // it will automatically update the UI with error messages if the validation fails.
+    // No need for further actions here in case it fails
+    if (!validator.validate()) {
+      return;
+    }
+
+    onGenerate(rule);
+  }
+
+  return (
+    <Validation>
+      {({ validator }) => (
+        <>
+          <DialogContent flex="0 0 auto" minHeight="400px">
+            {attempt.status === 'failed' && (
+              <Alert kind="danger" children={attempt.statusText} />
+            )}
+            <Box mb={4}>
+              <Text bold as="span" mt={1}>
+                Step 1
+              </Text>{' '}
+              - Assign IAM role to AWS resources
+              <Text mt={2}>
+                Every node using AWS IAM method to join your Teleport cluster
+                needs to be assigned an IAM role.
+              </Text>
+              <Text mt={1}>
+                If it doesn't already exist, create the IAM role "teleport_join"
+                and add it to all resources you wish to join your Teleport
+                cluster
+              </Text>
+              <Text mt={1}>
+                For more information, see documentation{' '}
+                <Link href="https://goteleport.com/docs/setup/guides/joining-nodes-aws/">
+                  here
+                </Link>
+                .
+              </Text>
+            </Box>
+            <Box mb={4}>
+              <Text bold as="span" mt={1}>
+                Step 2
+              </Text>{' '}
+              - Specify which nodes can join your Teleport cluster.
+              <Box mt={2}>
+                <FieldInput
+                  label="AWS Account ID"
+                  labelTip="nodes must match this AWS Account ID to join your Teleport cluster"
+                  autoFocus
+                  name="awsAccountId"
+                  onChange={e =>
+                    setRule({ ...rule, awsAccountId: e.target.value })
+                  }
+                  rule={requiredAwsAccountId}
+                  placeholder="111111111111"
+                  value={rule.awsAccountId}
+                  onKeyPress={e =>
```

Comment:
I think that these fields and the submit button should both be in a `form` so that you can rely on the native submit handling of a form. You can then attach a single `onsubmit` event handler to handle the submits.